### PR TITLE
Unit test for Openshifthelper

### PR DESCRIPTION
**Pull request by anandrkskd**
## Description
Increase coverage for OpenShiftHelper in scope of #207 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] ~I Added [CHANGELOG](../CHANGELOG.md) entry~
 - [x] I have implemented unit tests to cover my changes
 - [ ] ~I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly~
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] ~I tested my code in Kubernetes~
 - [ ] ~I tested my code in OpenShift~

### CODE REVIEW

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/OpenshiftHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class OpenshiftHelperTest {
+
+    @Rule
+    public KubernetesServer server = new KubernetesServer();
+//    OpenShiftMockServer mockServer = new OpenShiftMockServer(false);
```
Comment (manusa) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/OpenshiftHelperTest.java:44]: Remove dead code

**Comment by manusa** [comment]
The real purpose here would be to create **Unit tests**, as of now you are creating _Integration Tests_ since many of the tested parts + mocks are actually in Kuberentes-Client module.

It would be a better approach to Mock KuberentesClient class where necessary.

The main drawback of an integration test like this one is that if things change in Kubernetes Client (e.g. API endpoints hit to determine which type of client it is), our tests will start failing with no apparent reason. When implementing a unit test, what we want is to mock any public API method from our dependencies so that what we test is actually our code and not our dependency code.

**Comment by manusa** [comment]
As an example of the previous approach, test `testOpenShiftClientWithOpenShift` can be rewritten to 2 tests that will test all conditions:
```java
    @Test
    public void testOpenShiftClientWithAdaptableToOpenShift(@Mocked KubernetesClient kc, @Mocked OpenShiftClient oc) {
        // Given
        // @formatter:off
        new Expectations() {{
            kc.adapt(OpenShiftClient.class);
            result = oc;
        }};
        // @formatter:on
        //When
        OpenShiftClient result = OpenshiftHelper.asOpenShiftClient(kc);
        //Then
        assertNotNull(result);
    }

    @Test
    public void testOpenShiftClientWithOpenShift(@Mocked OpenShiftClient oc) {
        //When
        OpenShiftClient result = OpenshiftHelper.asOpenShiftClient(oc);
        //Then
        assertNotNull(result);
    }
```

You don't really care about the internals of KubernetesClient but what the public API of KubernetesClient returns. So in case any of the internals of KubernetesClient (or any other library) change in the future, this test will still be reliable and won't need changes. i.e. Test will only require changes if KubernetesClient public API changes.

File: jkube-kit/common/pom.xml [hunk 2]
```diff
@@ -135,6 +135,12 @@
       <groupId>com.networknt</groupId>
       <artifactId>json-schema-validator</artifactId>
     </dependency>
+      <dependency>
+          <groupId>io.fabric8</groupId>
+          <artifactId>kubernetes-server-mock</artifactId>
+          <version>4.10.2</version>
```
Comment (rohanKanojia) [inline jkube-kit/common/pom.xml:141]: I don't think you need to specify version here since it's covered by `<dependencyManagement>`

Comment (rohanKanojia) [reply jkube-kit/common/pom.xml:141]: anyway, If you are not going to mock Kubernetes API. You can remove this

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/OpenshiftHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+public class OpenshiftHelperTest {
+
+    @Rule
+    public KubernetesServer server = new KubernetesServer();
+
+    @Test
+    public void testAsOpenShiftClientWithNoOpenShift() {
+        // Given
+        KubernetesClient client = server.getClient();
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/OpenshiftHelperTest.java:49]: I think you should be able to use mocked `KubernetesClient` here also.

**Pull request merged**